### PR TITLE
Adds a visual effect to the Ravager's Ravage ability

### PR DESCRIPTION
**Pull request by lKiseki**
## About The Pull Request
ravagerravage

## Why It's Good For The Game
Looks good. Better-looking game is good.

## Changelog
:cl: Lewdcifer
imageadd: Added a visual effect to the Ravager's Ravage ability.
/:cl:

### CODE REVIEW

**Comment by MiniMeatwad** [comment]
Ravager buffed, now 2x more cool

**Comment by Lumipharon** [comment]
The ravage effect is handy, but it should more clearly cover the three tiles it actually hits. The gif shows it covering like 5 tiles which is misleading.

The effect for charge imo is fairly redundant since you can already see what's happening, and the effect is especially animeish

**Comment by lKiseki** [comment]
> The ravage effect is handy, but it should more clearly cover the three tiles it actually hits. The gif shows it covering like 5 tiles which is misleading.

I can try to tweak the ravage effect a little so it's smaller and occupies less screen size, but I can't do much about the other tiles it occupies. I changed it, so it should reflect the ability's range more accurately, but if I go much further it wouldn't look good.

ravagerravage
##
> The effect for charge imo is fairly redundant since you can already see what's happening, and the effect is especially animeish

In my opinion, a Ravager suddenly snapping to another side of the screen isn't exactly telling you what happened. It also doesn't look good. I daresay the new animation helps solve both of these concerns. I've tweaked the charge a little so it looks slightly better.

ravagercharge

**Review by Kuro020 — APPROVED** [review]

**Comment by FIXTHERADBOXPLEASE** [comment]
The style for the effects might be left field but it looks great IMO. Emphasizes speed and cutting-slashes, which is exactly what I think of with the Ravager.

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 1]
```diff
@@ -62,9 +64,27 @@
 	succeed_activate()
 
 	X.throw_at(A, RAV_CHARGEDISTANCE, RAV_CHARGESPEED, X)
+	activate_particles(TRUE, X.dir)
 
 	add_cooldown()
 
+/datum/action/xeno_action/activable/charge/proc/activate_particles(boolean = FALSE, direction)
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:71]: autodoc. Also why is that arg just called boolean? Make it something informative.

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 2]
```diff
@@ -62,9 +64,27 @@
...
+	if(!boolean)
+		QDEL_NULL(particle_holder)
+		return
+	particle_holder = new(get_turf(owner), /particles/ravager_charge)
+	addtimer(CALLBACK(src, .proc/activate_particles), 5) // Boolean is false by default, so this deactivates particles.
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:76]: Use QDEL_NULL_IN

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 3]
```diff
@@ -62,9 +64,27 @@
...
+	particle_holder.particles.rotation += dir2angle(direction)
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:77]: Doesn't using dir mean the effect won't be facing the right direction if you use this in a non cardinal direction?

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 4]
```diff
@@ -132,6 +167,27 @@
 	succeed_activate()
 	add_cooldown()
 
+/// Handles the activation and deactivation of particles, as well as their appearance.
+/datum/action/xeno_action/activable/ravage/proc/activate_particles(boolean = FALSE, direction)
+	if(!boolean)
+		QDEL_NULL(particle_holder)
+		return
+	particle_holder = new(owner, /particles/ravager_slash)
+	addtimer(CALLBACK(src, .proc/activate_particles), 5) // Boolean is false by default, so this deactivates particles.
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:176]: same stuff as charge.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm (hunk 3 above)
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:77]: Doesn't seem to be the case. I definitely tested non-cardinal directions, and everything seemed fine.

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm (hunk 1 above)
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:71]: I've never been good at naming anything in general. I'll call it toggle and hope that's fine.

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm (hunk 3 above)
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm]: This aged like milk. Which is to say, awfully.

**Review by TiviPlus — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 5]
```diff
@@ -132,6 +135,24 @@
 	succeed_activate()
 	add_cooldown()
 
+/// Handles the activation and deactivation of particles, as well as their appearance.
+/datum/action/xeno_action/activable/ravage/proc/activate_particles(direction)
```
[the remark this answers is not in the document]
Comment (lKiseki) [reply code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm]: Donezo.

**Pull request merged**